### PR TITLE
New Quartz host configs for clang14.0.6 and gcc12.1.1

### PR DESCRIPTION
**Pull request by CusiniM**
This PR adds new host-configs for quartz, using clang14 and gcc12 and removes old host-configs since those compilers are not available any more. 

LvArray PR: https://github.com/GEOS-DEV/LvArray/pull/282
TPLs PR: https://github.com/GEOS-DEV/thirdPartyLibs/pull/216

### CODE REVIEW

**Comment by wrtobin** [comment]
Just a heads up: the .gitlab-ci.yml nightly builds used the legacy quartz-host configs, so that process will fail once this is merged. Should just be a one-line change for the host-config variable though.

**Comment by bmhan12** [comment]
> Just a heads up: the .gitlab-ci.yml nightly builds used the legacy quartz-host configs, so that process will fail once this is merged.

It's currently failing with the legacy host-config, so this new host-config would fix it.

> Should just be a one-line change for the host-config variable though.

This section needs to be updated with the new host-config, and the quartz test should be fixed:
https://github.com/GEOS-DEV/GEOS/blob/b177414346e1da3ca8bc5d64a13a8e9f9d3d40fb/.gitlab-ci.yml#L118-L123

**Comment by wrtobin** [comment]
True!

"will fail for an entirely different reason" 

is the most correct phrasing, lol

File: host-configs/LLNL/quartz-base.cmake [hunk 1]
```diff
@@ -29,10 +29,14 @@ set(ENABLE_OPENMP ON CACHE BOOL "")
 set(ENABLE_PVTPackage ON CACHE BOOL "")
 set(ENABLE_PETSC OFF CACHE BOOL "Enables PETSc." FORCE)
 
+# PYGEOSX
+set(ENABLE_PYGEOSX ON CACHE BOOL "")
+set(Python3_EXECUTABLE /usr/gapps/GEOSX/thirdPartyLibs/python/quartz-gcc-python/python/bin/python3 CACHE PATH "")
```
Comment (cssherman) [inline host-configs/LLNL/quartz-base.cmake:34]: Sometimes we also need to add this line as well for cmake to find everything:

`set(Python3_ROOT_DIR /usr/gapps/GEOSX/thirdPartyLibs/python/quartz-gcc-python/python CACHE PATH "")`

**Review by rrsettgast — APPROVED** [review]

**Comment by rrsettgast** [comment]
> > Just a heads up: the .gitlab-ci.yml nightly builds used the legacy quartz-host configs, so that process will fail once this is merged.
> 
> It's currently failing with the legacy host-config, so this new host-config would fix it.
> 
> > Should just be a one-line change for the host-config variable though.
> 
> This section needs to be updated with the new host-config, and the quartz test should be fixed:
> 
> https://github.com/GEOS-DEV/GEOS/blob/b177414346e1da3ca8bc5d64a13a8e9f9d3d40fb/.gitlab-ci.yml#L118-L123

I think I addressed this properly in the last commit.

File: host-configs/LLNL/quartz-base.cmake (hunk 1 above)
Comment (rrsettgast) [reply host-configs/LLNL/quartz-base.cmake]: This is added now.

**Pull request merged**